src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -93,6 +101,51 @@ public async Task<DataTable> ExecuteQuery(string query, string stampName, string
             return await _kustoClient.ExecuteQueryAsync(Helpers.MakeQueryCloudAgnostic(_kustoMap, query), _kustoMap.MapCluster(cluster) ?? cluster, _kustoMap.MapDatabase(_configuration.DBName) ?? _configuration.DBName, requestId, operationName);
         }
 
+        public async Task<DataTable> ExecuteQueryOnAllAppAppServiceClusters(string query, string requestId = null, string operationName = null)
+        {
+            if (string.IsNullOrWhiteSpace(operationName))
+            {
+                throw new ArgumentNullException(nameof(operationName), "OperationName cannot be empty. Please supply a name to identify the query.");
+            }
+            if (string.IsNullOrWhiteSpace(query))
+            {
+                throw new ArgumentNullException(nameof(query), "Query cannot be empty. Please supply a query to execute.");
+            }
+            else
+            {
+                List<Task<DataTable>> queryTask = new List<Task<DataTable>>();
+
+                if(IsPublicCloud)
+                {
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-bay-153", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|WestUS"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-blu-189", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|EastUS"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-dm1-187", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|CentralUS"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-am2-329", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|WestEurope"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-db3-169", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|NorthEurope"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-hk1-029", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|EastAsia"));
+                }
+                else
+                {
+                    queryTask.Add(ExecuteQuery(query, DataProviderConstants.FakeStampForAnalyticsCluster, null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|{_configuration.CloudDomain}"));
+                }
+
+                var queryResult = await Task.WhenAll(queryTask.ToArray());
+                DataTable mergedTable = new DataTable();
+                foreach (DataTable dt in queryResult)
+                {
+                    if (dt.Rows.Count > 0)
+                    {
+                        if(dt.Rows.Count > 3000)
```

Comment:
Hey @nmallick1 , how did you decide 3000 would be a threshold? Can we also add a configuration value for this and you can make a default value of 3000 in the app settings

Reply:
I did a show query analysis for all queries that contain All for the past 7 days, the max rows returned was around 2K. I'm thinking 3K should be fine and anything more is actually too much.

Do you think think we'd need to reconfigure this frequently? I know it's cleaner but do we really need a config setting for this? 

If we do decide to introduce a rows limit via config setting, it should apply to all queries executed against Kusto not just this specific code path.